tensorflow/lite/micro/kernels/transpose_conv.cc

```diff
@@ -259,17 +273,38 @@ TfLiteStatus Eval(TfLiteContext* context, TfLiteNode* node) {
     case kTfLiteInt16: {
       std::int64_t* scratch_buffer = static_cast<int64_t*>(
           context->GetScratchBuffer(context, data.scratch_buffer_index));
-      reference_integer_ops::TransposeConv(
-          data.params, data.per_channel_output_multiplier,
-          data.per_channel_output_shift, tflite::micro::GetTensorShape(input),
-          tflite::micro::GetTensorData<int16_t>(input),
-          tflite::micro::GetTensorShape(filter),
-          tflite::micro::GetTensorData<int8_t>(filter),
-          tflite::micro::GetTensorShape(bias),
-          tflite::micro::GetTensorData<std::int64_t>(bias),
-          tflite::micro::GetTensorShape(output),
-          tflite::micro::GetTensorData<int16_t>(output),
-          tflite::micro::GetTensorShape(nullptr), nullptr, scratch_buffer);
+      // Handle models with int16 biases by converting to int64 on the fly.
+      if (bias->type == kTfLiteInt16) {
+        std::int64_t* bias_converted_buffer =
+            static_cast<int64_t*>(context->GetScratchBuffer(
+                context, data.bias_converted_buffer_index));
+        for (int i = 0; i < tflite::micro::GetTensorShape(bias).FlatSize();
+             i++) {
+          bias_converted_buffer[i] = bias->data.i16[i];
+        }
```

Comment:
the complexity in the inference code makes me wonder if we should do something else.

Reply:
As this is a temporary workaround, I would prefer to keep it as-is. I cannot think of a simpler immediate solution. Model reworking would be both more complex and likely throwaway work. I'm open to suggestions, however.